app/src/main/java/com/pajato/argus/SearchActivity.kt

```diff
@@ -18,15 +24,38 @@ class SearchActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_search)
+        setSupportActionBar(toolbar)
     }
 
-    fun save(view: View) {
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // Inflate the menu and color the check icon to be the correct color
+        menuInflater.inflate(R.menu.save_menu, menu)
+        val drawable = menu.getItem(0).icon
+        drawable.mutate()
+        val color = ContextCompat.getColor(applicationContext, android.R.color.white)
+        drawable.setColorFilter(color, PorterDuff.Mode.SRC_ATOP)
+        return true
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // Add the auto-fill resources to the networks input.
+        val networks: List<String> = resources.getStringArray(R.array.networks).toList()
+        val arrayAdapter = ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, networks)
+        network.setAdapter(arrayAdapter)
+        // Force the activity to focus on the name input
+        name.requestFocus()
+        val imm: InputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.showSoftInput(name, InputMethodManager.SHOW_IMPLICIT)
+    }
+
+    fun save(menuItem: MenuItem) {
```

Comment:
After looking at this for a while I opted to go without a parameter (removing the onClick from the XML) and invoking the save() from the onOptionsSelected() callback.  This strikes me as more idiomatic.